### PR TITLE
LPS-154705 Upload Fragment Image -- Cannot Edit Image and Save

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
- LPS-154705 You can upload a file or edit the image on the same modal, so you must differentiate both behaviors
- LPS-154705 extract methods and reorder
- LPS-154705 extract methods to common class to reuse on several modules
- LPS-154705 baseline
- LPS-154705 use new class on StyleBookEntryPreviewUploadFileEntryHandler
- LPS-154705 use new class on FragmentEntryPreviewUploadFileEntryHandler
- LPS-154705 use new class on FragmentCollectionResourceUploadFileEntryHandler

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
@adolfopa I you think thats the classes are well placed. I will follow or push another commits, making the same change on others places that we use the same process

**Comment by Hanlf** [comment]
ci:test:relevant

**Comment by Hanlf** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Review by adolfopa — CHANGES_REQUESTED** [review]
The three upload handlers are mostly identical, for what I've seen, they only differ in the permission check. Instead of a new interface, why not create a new `ImageEditorUploadHandler` that contains all duplicated logic and a template method to check permissions, and let each of the upload handlers extend that, customizing the permission check.

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

File: modules/apps/upload/upload-api/src/main/java/com/liferay/upload/ImageEditorUploadHandler.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.kernel.util.WebKeys;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * @author Alicia García
+ */
+public abstract class ImageEditorUploadHandler
```
Comment (adolfopa) [inline modules/apps/upload/upload-api/src/main/java/com/liferay/upload/ImageEditorUploadHandler.java:35]: Can you name this class `BaseImageEditorUploadHandler` for consistency with other similar clases?

**Review by adolfopa — CHANGES_REQUESTED** [review]

**Comment by AliciaGarciaGarcia** [comment]
@adolfopa : class renamed

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Review by adolfopa — APPROVED** [review]

**Pull request closed without merging**